version: "3.7"

services:

  nginx:
    image: nginx:1.19.5
    ports:
      - 80:80
    networks:
      - gis_5ban_network
    volumes:
      - /home/ubuntu/nginx.conf:/etc/nginx/nginx.conf
      - gis_5ban_static:/data/static/
      - gis_5ban_media:/data/media/
      - gis_log:/var/log/nginx/

  goaccess:
    image: allinurl/goaccess:1.5.2
    volumes:
    - gis_log:/var/log/nginx/
    ports:
    - 7890:7890
    command: /var/log/nginx/gis_access.log -o /var/log/nginx/report.html --log-format=COMBINED --real-time-html

  django:
    image: gis_5ban_django:5
    networks:
      - gis_5ban_network
    volumes:
      - gis_5ban_static:/home/gis_5ban_1/staticfiles/
      - gis_5ban_media:/home/gis_5ban_1/media/
    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD

  mariadb:
    image: mariadb:10.5
    networks:
      - gis_5ban_network
    volumes:
      - gis_mariadb:/var/lib/mysql
    secrets:
      - MARIADB_USER
      - MARIADB_PASSWORD
      - MARIADB_ROOT_PASSWORD
    environment:
      MARIADB_DATABASE: django
      MARIADB_USER_FILE: /run/secrets/MARIADB_USER
      MARIADB_PASSWORD_FILE: /run/secrets/MARIADB_PASSWORD
      MARIADB_ROOT_PASSWORD_FILE: /run/secrets/MARIADB_ROOT_PASSWORD


networks:
  gis_5ban_network:

volumes:
  gis_5ban_static:
  gis_5ban_media:
  gis_mariadb:
  gis_log:

secrets:
  DJANGO_SECRET_KEY:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true




